src/openrct2/ride/Track.cpp

```diff
@@ -2222,120 +2190,170 @@ bool track_element_has_speed_setting(uint8_t trackType)
+void TrackElement::SetTrackType(uint8_t newType)
 {
-    tileElement->properties.track.sequence = tile_element_get_track_sequence(tileElement) | ((speed >> 1) << 4);
+    trackType = newType;
 }
 
-bool tile_element_is_taking_photo(const rct_tile_element* tileElement)
+uint8_t TrackElement::GetSequenceIndex() const
 {
-    return (tileElement->properties.track.sequence & MAP_ELEM_TRACK_SEQUENCE_TAKING_PHOTO_MASK) != 0;
+    return sequence & MAP_ELEM_TRACK_SEQUENCE_SEQUENCE_MASK;
 }
 
-void tile_element_set_onride_photo_timeout(rct_tile_element* tileElement)
+void TrackElement::SetSequenceIndex(uint8_t newSequenceIndex)
 {
-    tileElement->properties.track.sequence &= MAP_ELEM_TRACK_SEQUENCE_SEQUENCE_MASK;
-    tileElement->properties.track.sequence |= (3 << 4);
+    sequence &= ~MAP_ELEM_TRACK_SEQUENCE_SEQUENCE_MASK;
+    sequence |= (newSequenceIndex & MAP_ELEM_TRACK_SEQUENCE_SEQUENCE_MASK);
 }
 
-void tile_element_decrement_onride_photo_timout(rct_tile_element* tileElement)
+uint8_t TrackElement::GetStationIndex() const
 {
-    // We should only touch the upper 4 bits, avoid underflow into the lower 4.
-    if (tileElement->properties.track.sequence & MAP_ELEM_TRACK_SEQUENCE_TAKING_PHOTO_MASK)
-    {
-        tileElement->properties.track.sequence -= (1 << 4);
-    }
+    return (sequence & MAP_ELEM_TRACK_SEQUENCE_STATION_INDEX_MASK) >> 4;
+}
+
+void TrackElement::SetStationIndex(uint8_t newStationIndex)
+{
+    sequence &= ~MAP_ELEM_TRACK_SEQUENCE_STATION_INDEX_MASK;
+    sequence |= (newStationIndex << 4);
+}
+
+uint8_t TrackElement::GetDoorAState() const
+{
+    return (colour & TRACK_ELEMENT_DOOR_A_MASK) >> 2;
+}
+
+uint8_t TrackElement::GetDoorBState() const
+{
+    return (colour & TRACK_ELEMENT_DOOR_B_MASK) >> 5;
+}
+
+uint8_t TrackElement::GetRideIndex() const
+{
+    return rideIndex;
+}
+
+void TrackElement::SetRideIndex(uint8_t newRideIndex)
+{
+    rideIndex = newRideIndex;
+}
+
+uint8_t TrackElement::GetColourScheme() const
+{
+    return colour & 0x3;
 }
 
-uint16_t track_element_get_maze_entry(const rct_tile_element* tileElement)
+void TrackElement::SetColourScheme(uint8_t newColourScheme)
 {
-    return tileElement->properties.track.maze_entry;
+    colour &= ~0x3;
+    colour |= (newColourScheme & 0x3);
 }
 
-uint8_t track_element_get_ride_index(const rct_tile_element* tileElement)
+bool TrackElement::HasCableLift() const
 {
-    return tileElement->properties.track.ride_index;
+    return colour & TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
 }
 
-void track_element_set_ride_index(rct_tile_element* tileElement, uint8_t rideIndex)
+void TrackElement::SetHasCableLift(bool on)
 {
-    tileElement->properties.track.ride_index = rideIndex;
+    colour &= ~TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
+    if (on)
+        colour |= TRACK_ELEMENT_COLOUR_FLAG_CABLE_LIFT;
+}
+
+bool TrackElement::IsInverted() const
+{
+    return colour & TRACK_ELEMENT_COLOUR_FLAG_INVERTED;
+}
+
+void TrackElement::SetInverted(bool inverted)
+{
+    if (inverted)
```

Comment:
You may be able to use the `^=` operator for this block.

Reply:
I prefer the current method for consistency.